File: 04-factorial/factorial.go
// Go program to compute factorial
// Created by M.V.Harish Kumar on 25-10-2023
package main

import "fmt"

func factorial(n int) float64 {
	if n <= 1 {
		return 1
	} else {
		return float64(n) * factorial(n-1)
	}
}

func main() {
	var n int
	fmt.Print("Enter value for `n`: ")
	fmt.Scanf("%d", &n)
	fmt.Printf("%d! = %f\n", n, factorial(n))
}

File: 04-factorial/factorial.exs
# Elixir program to compute factorial
# Created by M.V.Harish Kumar on 10-12-2023

defmodule Factorial do
  def fact(0, acc) do
    acc
  end

  def fact(n, acc) do
    fact(n-1, n*acc)
  end

  def fact(n) do
    fact(n, 1)
  end
end

{n, _} = IO.gets("Enter value for `n`: ") |> Integer.parse
ans = Factorial.fact(n)
IO.puts "#{n}! = #{ans}"
